src/app/routes/api/v3/payments/payments.routes.ts

```diff
@@ -7,14 +7,25 @@ export const PaymentsRouter = Router()
 PaymentsRouter.get('/stripe')
 
 /**
- * Returns the receipt URL to the user
- * @route GET /receipt/:formId/:submissionId
+ * Checks if the payment receipt is ready
+ * @route GET /payments/receipt/:formId/:submissionId/status
+ *
+ * @returns 200 if receipt URL exists
+ * @returns 404 if receipt URL does not exist
+ */
+PaymentsRouter.route(
+  '/receipt/:formId([a-fA-F0-9]{24})/:submissionId([a-fA-F0-9]{24})/status',
+).get(StripeController.checkPaymentReceiptStatus)
+
+/**
+ * Downloads the receipt pdf
+ * @route GET /receipt/:formId/:submissionId/download
  *
  * @returns 200 with receipt URL exists
  */
 PaymentsRouter.route(
-  '/receipt/:formId([a-fA-F0-9]{24})/:submissionId([a-fA-F0-9]{24})',
-).get(StripeController.getPaymentReceipt)
+  '/receipt/:formId([a-fA-F0-9]{24})/:submissionId([a-fA-F0-9]{24})/download',
```

Comment:
Let's add a TODO comment here that we should consider rate limiting this endpoint 🤔 . Please create an issue for that as well.

Reply:
[c8ea1be](https://github.com/opengovsg/FormSG/pull/5837/commits/c8ea1bead9d14ab779ecdda23d00f62ca154993c)

issue #5924

Reply:
Thanks @wanlingt , I added some rationale in the ticket.